### PR TITLE
feat!: allow serializing with capacity

### PR DESCRIPTION
**Pull request by tbrezot**
- add `<impl Serializable>::length()`
- add `Serializer::with_capaity()`
- remove `Serializer::value()` (useless)

BREAKING CHANGE: removing of `Serializer::value()`

**Note**: to be reviewed with [the similar MR](https://github.com/Cosmian/cover_crypt/pull/60) on CoverCrypt

### CODE REVIEW

File: src/bytes_ser_de.rs [hunk 1]
```diff
@@ -108,19 +122,6 @@ impl<'a> Deserializer<'a> {
         Ok(buf)
     }
 
-    /// Reads all the remaining bytes.
-    #[inline]
-    pub fn value(&mut self) -> Result<Vec<u8>, CryptoCoreError> {
```
Comment (ThibsG) [inline src/bytes_ser_de.rs:113]: shouldn't we move to 5.0.0 as we are removing public API?

Comment (tbrezot) [reply src/bytes_ser_de.rs:113]: Yes we should :/

Comment (tbrezot) [reply src/bytes_ser_de.rs:113]: We now use `Unreleased` in the changelog and don't change the version when pushing to `develop`. The version is fixed when pushing to `main`.

File: src/bytes_ser_de.rs [hunk 2]
```diff
@@ -10,16 +10,23 @@ pub trait Serializable: Sized {
     /// Error type returned by the serialization.
     type Error: From<CryptoCoreError>;
 
+    /// Retrieves the length of the serialized object if it can be known.
+    ///
+    /// This length will be used to initialize the `Serializer` with the
+    /// correct capacity in `try_to_bytes()`. If `None` is provided, 0 will be
```
Comment (ThibaudDauce) [inline src/bytes_ser_de.rs:16]: `None` cannot be provided here, no?

Comment (tbrezot) [reply src/bytes_ser_de.rs:16]: Indeed, I changed the code.

**Review by ThibaudDauce — APPROVED** [review]

**Pull request merged**